src/backend/columnar/columnar_tableam.c

```diff
@@ -627,7 +705,8 @@ columnar_relation_copy_for_cluster(Relation OldHeap, Relation NewHeap,
 	if (OldIndex != NULL || use_sort)
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("indexes not supported for columnar tables")));
+						errmsg("clustering columnar tables using indexes is "
+							   "not supported")));
```

Comment:
Don't have a reason for not implementing that, just skipped not to extend the scope for phase-2, but we can still spend some time on that if time permits.